webtau-core/src/main/java/com/twosigma/webtau/expectation/equality/handlers/HandlerMessages.java

```diff
@@ -16,22 +16,55 @@
 
 package com.twosigma.webtau.expectation.equality.handlers;
 
+import com.twosigma.webtau.expectation.equality.CompareToComparator;
+
 import static com.twosigma.webtau.utils.TraceUtils.renderValueAndType;
 
 class HandlerMessages {
     private static final String ACTUAL_PREFIX = "  actual: ";
     private static final String EXPECTED_PREFIX = "expected: ";
 
-    static String renderActualExpected(Object actual, Object expected) {
+    static String renderActualExpected(CompareToComparator.AssertionMode assertionMode, Object actual, Object expected) {
         return ACTUAL_PREFIX + renderValueAndType(actual) + "\n" +
-                EXPECTED_PREFIX + renderValueAndType(expected);
+                expected(assertionMode, renderValueAndType(expected));
     }
 
-    static String renderActualExpected(Object convertedActual, Object convertedExpected,
+    static String renderActualExpected(CompareToComparator.AssertionMode assertionMode,
+                                       Object convertedActual, Object convertedExpected,
                                        Object actual, Object expected) {
         return ACTUAL_PREFIX + renderValueAndType(convertedActual) +
                 "(before conversion: " + renderValueAndType(actual) + ")\n" +
-                EXPECTED_PREFIX + renderValueAndType(convertedExpected) +
-                "(before conversion: " + renderValueAndType(expected) + ")";
+                expected(assertionMode, renderValueAndType(convertedExpected) +
+                "(before conversion: " + renderValueAndType(expected) + ")");
+    }
+
+    static String expected(CompareToComparator.AssertionMode assertionMode, Object expected) {
+        return expected(EXPECTED_PREFIX, assertionMode, expected);
+    }
+
+    static String expected(String prefix, CompareToComparator.AssertionMode assertionMode, Object expected) {
+        String expectedMsg = prefix;
+
+        switch (assertionMode) {
+            case NOT_EQUAL:
```

Comment:
minor nit to move this out as a function to use `returns` and avoid extra spaces at the end.

Reply:
I'll move it into a function, it's nicer than this large inline switch.  However, still need to add spaces somewhere and can't just prepend the expected text with a space for the equality case.